src/tools/kcSanitize/KcSanitizer.ts

```diff
@@ -0,0 +1,49 @@
+import { KcSanitizerPolicy } from "keycloakify/tools/kcSanitize/KcSanitizerPolicy";
+import { decode } from "html-entities";
+
+// implementation of keycloak java sanitize method ( KeycloakSanitizerMethod )
+// https://github.com/keycloak/keycloak/blob/8ce8a4ba089eef25a0e01f58e09890399477b9ef/services/src/main/java/org/keycloak/theme/KeycloakSanitizerMethod.java#L33
+export class KcSanitizer {
+    private static HREF_PATTERN = /\s+href="([^"]*)"/g;
+
+    public static sanitize(html: string | null): string {
+        if (html == null) {
+            throw new Error("Cannot escape null value.");
+        }
+        if (html === "") return "";
+
+        html = this.decodeHtmlFull(html);
+        const sanitized = KcSanitizerPolicy.sanitize(html);
+        return this.fixURLs(sanitized);
+    }
+
+    private static decodeHtmlFull(html: string): string {
+        return decode(html);
```

Comment:
@garronej Thanks for the input. I added a lazy import so that we use html-entities on the server and textarea on the client. I also refactored the tests to work in both cases, in case SSR is added in the future. Does it look good now?
I agree with you that it’s a lot of work for a small feature. It could have been much easier to implement using dompurify's default settings, but I wanted to emulate the exact same code Keycloak uses, so if they change it in the future, we can adjust our code accordingly.